tools/run_tests/jobset.py

```diff
@@ -229,15 +226,12 @@ def state(self, update_cache):
       self._tempfile.seek(0)
       stdout = self._tempfile.read()
       filtered_stdout = _filter_stdout(stdout)
```

Comment:
Good point. Can we at least do some cleanup and unify the truncating logic and perhaps add a todo?
One solution that sounds reasonable to me:
1.  add the max_result_size in the job/jobspec/jobset/.. configuration, truncation should be done upon job result creation.
2. the exact way of filtering output depends on the report output format (xml, html, ...) so filtering should be done in the report module, not here.